test/library/packages/UnitTest/launcher.chpl

```diff
@@ -0,0 +1,142 @@
+/*
+  This launcher will be used to run the test files and catch the halts.
+  You can specify a different directory name using the `--dir` 
+  You can set whether or not to descend recursively into 
+  subdirectories (defaults to true) using `--subdir`
+*/
+use FileSystem;
+use Spawn;
+use Path;
+use TestResult;
+config const subdir: string = "true";
+config const keepExec: string = "false";
+
+/*Docs: Todo*/
+proc run_and_log(executable,fileName, ref result,skipId = 0) {
+  var separator1 = result.separator1,
+      separator2 = result.separator2;
+  var testName: string,
+      flavour: string,
+      line: string,
+      tempString: string;
+  var curIndex = 0;
+  var sep1Found = false,
+      haltOccured = false;
+  
+  var exec = spawn(["./"+executable,"--skipId",skipId:string], stdout = PIPE, stderr = PIPE); //Executing the file
+  //std output pipe
+  while exec.stdout.readline(line) {
+    if line.strip() == separator1 then sep1Found = true;
+    else if line.strip() == separator2 && sep1Found {
+      if flavour == "ERROR" then result.addError(testName, tempString);
```

Comment:
If we expect this list of conditions to grow overtime, we may want to refactor into a switch statement.

Reply:
Yes, that is a better way of coding and looks clean as well. Just updated it. Thanks.